pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -107,190 +114,130 @@ type KubeSchedulerLeaderElectionConfiguration struct {
-}
-
-type CloudProviderConfiguration struct {
-	// Name is the provider for cloud services.
-	Name string `json:"cloudProvider"`
-	// cloudConfigFile is the path to the cloud provider configuration file.
-	CloudConfigFile string `json:"cloudConfigFile"`
+	NodeStatusUpdateFrequency metav1.Duration `json:"nodeStatusUpdateFrequency"`
 }
 
-type GenericComponentConfiguration struct {
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+type GenericControllerManagerConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+	
 	// minResyncPeriod is the resync period in reflectors; will be random between
 	// minResyncPeriod and 2*minResyncPeriod.
 	MinResyncPeriod metav1.Duration `json:"minResyncPeriod"`
-	// contentType is contentType of requests sent to apiserver.
-	ContentType string `json:"contentType"`
-	// kubeAPIQPS is the QPS to use while talking with kubernetes apiserver.
-	KubeAPIQPS float32 `json:"kubeAPIQPS"`
-	// kubeAPIBurst is the burst to use while talking with kubernetes apiserver.
-	KubeAPIBurst int32 `json:"kubeAPIBurst"`
+	// ClientConnection specifies the kubeconfig file and client connection
+	// settings for the proxy server to use when communicating with the apiserver.
+	ClientConnection apimachineryconfigv1alpha1.ClientConnectionConfiguration `json:"clientConnection"`
 	// How long to wait between starting controller managers
 	ControllerStartInterval metav1.Duration `json:"controllerStartInterval"`
 	// leaderElection defines the configuration of leader election client.
 	LeaderElection apiserverconfigv1alpha1.LeaderElectionConfiguration `json:"leaderElection"`
-}
-
-type KubeCloudSharedConfiguration struct {
 	// port is the port that the controller-manager's http service runs on.
 	Port int32 `json:"port"`
 	// address is the IP address to serve on (set to 0.0.0.0 for all interfaces).
 	Address string `json:"address"`
 	// useServiceAccountCredentials indicates whether controllers should be run with
 	// individual service account credentials.
 	UseServiceAccountCredentials bool `json:"useServiceAccountCredentials"`
+	// Controllers is the list of controllers to enable or disable
+	// '*' means "all enabled by default controllers"
+	// 'foo' means "enable 'foo'"
+	// '-foo' means "disable 'foo'"
+	// first item for a particular name wins
+	Controllers []string `json:"controllers"`
+	// DebuggingConfiguration holds configuration for Debugging related features.
+	Debugging apiserverconfigv1alpha1.DebuggingConfiguration `json:"debugging"`
+}
+
+type KubeCloudSharedConfiguration struct {
+	// CloudProviderConfiguration holds configuration for CloudProvider related features.
+	CloudProvider CloudProviderConfiguration `json:"cloudProvider"`
+	// externalCloudVolumePlugin specifies the plugin to use when cloudProvider is "external".
+	// It is currently used by the in repo cloud providers to handle node and volume control in the KCM.
+	ExternalCloudVolumePlugin string `json:"externalCloudVolumePlugin"`
```

Comment:
`ExternalCloudVolumePlugin` not belong to `cloud-controller-manager` now,  should not place in `KubeCloudSharedConfiguration`. WDYT